symbol.go

```diff
@@ -0,0 +1,99 @@
+package protocol
+
+type RangeBasedDocumentSymbol struct {
+	// ID is the range ID associated with this symbol.
+	ID       uint64                      `json:"id"`
+	Children []*RangeBasedDocumentSymbol `json:"children,omitempty"`
+}
+
+type DocumentSymbolResult struct {
+	Vertex
+	Result []*RangeBasedDocumentSymbol `json:"result"` // TODO: lsp.DocumentSymbol
+}
+
+func NewDocumentSymbolResult(id uint64, result []*RangeBasedDocumentSymbol) DocumentSymbolResult {
+	return DocumentSymbolResult{
+		Vertex: Vertex{
+			Element: Element{
+				ID:   id,
+				Type: ElementVertex,
+			},
+			Label: VertexDocumentSymbolResult,
+		},
+		Result: result,
+	}
+}
+
+type DocumentSymbolEdge struct {
+	Edge
+	InV  uint64 `json:"inV"`
+	OutV uint64 `json:"outV"`
+}
+
+func NewDocumentSymbolEdge(id, inV, outV uint64) DocumentSymbolEdge {
+	return DocumentSymbolEdge{
+		Edge: Edge{
+			Element: Element{
+				ID:   id,
+				Type: ElementEdge,
+			},
+			Label: EdgeTextDocumentDocumentSymbol,
+		},
+		OutV: outV,
+		InV:  inV,
+	}
+}
+
+type SymbolData struct {
```

Comment:
Can we get comments on each of these fields? Looks like these are still non-standard.

Reply:
I'm also unsure what in the spec this corresponds to.

Reply:
Added a docstring. It's not an extension, it's just the symbol-related embedded fields of the RangeTag struct, which corresponds to the DeclarationTag described in the spec: https://microsoft.github.io/language-server-protocol/specifications/lsif/0.5.0/specification/#documentSymbol

Reply:
Looks like the `Tags` field is missing from the spec, though? Is that part a non-standard extension?

Reply:
Yes, I just updated the docstring to mention https://github.com/microsoft/language-server-protocol/issues/1209, which proposes adding `SymbolTag[]` as a field, so that it mirrors the information present in `lsp.DocumentSymbol`.